app/controllers/api/v1/customer_payment_method_controller.rb

```diff
@@ -0,0 +1,77 @@
+module Api
+  module V1
+    class CustomerPaymentMethodController < Api::V1::ApiController
+      def index
+        @customer_payment_method = CustomerPaymentMethod.where(company: @company)
+
+        render status: :ok, json: success_json
+      end
+
+      def show
+        @customer_payment_method = CustomerPaymentMethod.find_by(token: params[:id], company: @company)
+        raise ActiveRecord::RecordNotFound if @customer_payment_method.nil?
+
+        render status: :ok, json: success_json
+      end
+
+      def create
+        @payment_method = find_payment_method
+        @customer = Customer.find_by(token: customer_payment_method_params[:customer_token], company: @company)
+
+        @customer_payment_method = CustomerPaymentMethod.new(
+          payment_method: @payment_method, customer: @customer, company: @company
+        )
+        @customer_payment_method.add_credit_card(credit_card_params) if @payment_method&.credit_card?
+
+        return render status: :created, json: success_json if @customer_payment_method.save
+
+        render status: :unprocessable_entity, json: error_json
```

Comment:
Para evitar desperdiçar processamento. O que acha de mover a validação para o início do método? 🤔 

Ex:
```suggestion
        @payment_method = find_payment_method
        return render { status: :unprocessable_entity, json: error_json } unless @payment_method&.credit_card?

        @customer = Customer.find_by(token: customer_payment_method_params[:customer_token], company: @company)

        @customer_payment_method = CustomerPaymentMethod.new(
          payment_method: @payment_method, customer: @customer, company: @company
        )
        @customer_payment_method.add_credit_card(credit_card_params)

        render status: :created, json: success_json if @customer_payment_method.save
```